test/com/google/javascript/jscomp/AmbiguatePropertiesTest.java

```diff
@@ -301,6 +303,78 @@ public void testQuotedPrototypeProperty() {
              "bar['getA']();");
   }
 
+  public void testObjectDefineProperties() {
+    String js = LINE_JOINER.join(
+        "/** @constructor */ var Bar = function(){};",
+        "Bar.prototype.bar = 0;",
+        "/** @struct @constructor */ var Foo = function() {",
+        "  this.bar_ = 'bar';",
+        "};",
+        "/** @type {?} */ Foo.prototype.bar;",
+        "Object.defineProperties(Foo.prototype, {",
+        "  bar: {",
+        "    configurable: true,",
+        "    enumerable: true,",
+        "    /** @this {Foo} */ get: function() { return this.bar_;},",
+        "    /** @this {Foo} */ set: function(value) { this.bar_ = value; }",
+        "  }",
+        "});");
+
+    String result = LINE_JOINER.join(
+        "/** @constructor */ var Bar = function(){};",
+        "Bar.prototype.a = 0;",
+        "/** @struct @constructor */ var Foo = function() {",
+        "  this.b = 'bar';",
+        "};",
+        "/** @type {?} */ Foo.prototype.a;",
+        "Object.defineProperties(Foo.prototype, {",
+        "  a: {",
+        "    configurable: true,",
+        "    enumerable: true,",
+        "    /** @this {Foo} */ get: function() { return this.b;},",
+        "    /** @this {Foo} */ set: function(value) { this.b = value; }",
+        "  }",
+        "});");
+
+    test (js, result);
+  }
+
+  public void testObjectDefinePropertiesQuoted() {
+    String js = LINE_JOINER.join(
+        "/** @constructor */ var Bar = function(){};",
+        "Bar.prototype.bar = 0;",
+        "/** @struct @constructor */ var Foo = function() {",
+        "  this.bar_ = 'bar';",
+        "};",
+        "/** @type {?} */ Foo.prototype['bar'];",
+        "Object.defineProperties(Foo.prototype, {",
+        "  'bar': {",
```

Comment:
If we make this quoted prop 'a', we'll get a test of that new case.

After that can you squash the commits into one?

Other than that, LGTM